specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2020-02-07-preview/hanaonazure.json

```diff
@@ -545,37 +617,39 @@
           "readOnly": true,
           "type": "string",
           "description": "The version of the payload running in the Collector VM"
-        },
-        "providerInstances": {
+        }
+      },
+      "description": "Describes the properties of a SAP monitor."
+    },
+    "ProviderInstanceListResult": {
+      "properties": {
+        "value": {
           "type": "array",
           "items": {
             "$ref": "#/definitions/ProviderInstance"
           },
-          "description": "The list of monitor provider instances."
+          "description": "The list of provider instances."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of provider instances."
         }
       },
-      "description": "Describes the properties of a SAP monitor."
+      "description": "The response from the List provider instances operation."
     },
     "ProviderInstance": {
       "properties": {
-        "id": {
-          "readOnly": true,
-          "type": "string",
-          "description": "Resource ID"
-        },
-        "name": {
-          "type": "string",
-          "description": "The name of the provider instance."
-        },
-        "type": {
-          "type": "string",
-          "description": "The provider instance type."
-        },
```

Comment:
Why is there no modelling of the properties of the provider instance?  This is going to make writing tools to consume this type very difficult, both in SDKs and for customers consuming the API directly

Reply:
This is by design. the properties field is a JSON string and the schema will be determined by the type field.

We wanted to make the API flexible. Also, other teams will be implementing their own providers and it will be up to them what their JSON schema will be.